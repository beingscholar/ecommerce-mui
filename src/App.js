import React from "react";
import "./App.scss";
import { NotificationContainer } from "react-notifications";
import "react-notifications/lib/notifications.css";

import LoadingIndicator from "./components/layouts/LoadingIndicator";
import Main from "./components/Main";

import Amplify from "aws-amplify";

import "@aws-amplify/ui/dist/style.css";

Amplify.configure({
  Auth: {
    // REQUIRED - Amazon Cognito Region
    region: "ap-southeast-1",
    // OPTIONAL - Amazon Cognito User Pool ID
    userPoolId: "ap-southeast-1_fPG3z00th",
    // OPTIONAL - Amazon Cognito Web Client ID (26-char alphanumeric string)
    userPoolWebClientId: "1pgoli69kjl90l7sm6q4nmstrc",
  },
});
//Amplify.configure(awsconfig);

function App() {
  return (
    <div>
      <React.Fragment>
        <LoadingIndicator />
      </React.Fragment>
      <Main />
      <React.Fragment>
        <NotificationContainer />
      </React.Fragment>
    </div>
  );
}

export default App;
