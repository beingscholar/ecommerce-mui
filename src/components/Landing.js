import { Box, Button } from "@material-ui/core";
import React, { useEffect, useState } from "react";
import {
  Route,
  BrowserRouter as Router,
  Link as RouterLink,
  Switch,
} from "react-router-dom";

import ArrowBackIosIcon from "@material-ui/icons/ArrowBackIos";
import ArrowForwardIosIcon from "@material-ui/icons/ArrowForwardIos";
import Avatar from "@material-ui/core/Avatar";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Container from "@material-ui/core/Container";
import Divider from "@material-ui/core/Divider";
import DraftsIcon from "@material-ui/icons/Drafts";
import Grid from "@material-ui/core/Grid";
import InboxIcon from "@material-ui/icons/Inbox";
import Link from "@material-ui/core/Link";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import MediaQuery from "react-responsive";
import Rating from "@material-ui/lab/Rating";
import Slider from "react-slick";
import Typography from "@material-ui/core/Typography";
import { createBrowserHistory as history } from "history";
import { makeStyles } from "@material-ui/core/styles";
import { trackPromise } from "react-promise-tracker";

const url =
  "http://myproject-alb-692769319.ap-southeast-1.elb.amazonaws.com/products";

const useStyles = makeStyles((theme) => ({
  navLink: {
    margin: theme.spacing(2),
    fontSize: "1rem",
  },
  title: {
    flexGrow: 1,
    display: "none",
    [theme.breakpoints.up("sm")]: {
      display: "block",
    },
  },
  authLinks: {
    position: "relative",

    marginLeft: 0,
    width: "100%",
    [theme.breakpoints.up("sm")]: {
      marginLeft: theme.spacing(1),
      width: "auto",
    },
  },
  heroContent: {
    backgroundColor: theme.palette.background.paper,
    padding: theme.spacing(8, 0, 6),
  },
  heroButtons: {
    marginTop: theme.spacing(4),
  },
  cardGrid: {
    paddingTop: theme.spacing(8),
    paddingBottom: theme.spacing(8),
  },
  card: {
    height: "100%",
    display: "flex",
    flexDirection: "column",
  },
  cardMedia: {
    paddingTop: "56.25%", // 16:9
  },
  cardContent: {
    flexGrow: 1,
  },
  footer: {
    backgroundColor: theme.palette.background.paper,
    padding: theme.spacing(6),
  },
}));

export default function Landing() {
  const classes = useStyles();
  const [products, setProducts] = useState([]);

  useEffect(() => {
    trackPromise(
      fetch(url)
        .then((response) => {
          return response.json();
        })
        .then((data) => {
          setProducts(data.products);
        })
        .catch((error) => {
          alert(error);
        })
    );
  }, []);

  if (products) {
    var bannerList = products.map((product) => {
      return (
        <Card key={product.productId} className="product-card">
          <CardMedia
            image={product.imageUrl} /* change to product.imageUrl */
            title="Image title"
          />

          <CardContent>
            <Typography component="h2">{product.productDescription}</Typography>
            <CardActions>
              <Button
                variant="contained"
                color="primary"
                disableElevation
                type="button"
              >
                Shop Now
              </Button>
            </CardActions>
          </CardContent>
        </Card>
      );
    });
  }

  if (products) {
    var productList = products.map((product) => {
      return (
        <Card key={product.productId} className="product-card">
          <CardActionArea
            component={RouterLink}
            to={"/products/" + product.productId}
          >
            <CardMedia
              image={product.imageUrl} /* change to product.imageUrl */
              title="Image title"
            />
          </CardActionArea>

          <CardContent>
            <Typography component="h4">{product.productName}</Typography>
            <Typography component="h5">
              {product.currency} {product.price}
            </Typography>
            <Typography className="description">
              {product.productDescription}
            </Typography>
            <Typography className="product-rating">
              <Rating
                name="half-rating-read"
                defaultValue={2.5}
                precision={0.5}
                readOnly
              />
              (55)
            </Typography>
          </CardContent>
        </Card>
      );
    });
  }

  var productListSlider = {
    dots: false,
    infinite: false,
    speed: 500,
    slidesToShow: 5,
    slidesToScroll: 1,
    initialSlide: 0,
    prevArrow: (
      <Box component="div">
        <ArrowBackIosIcon />
      </Box>
    ),
    nextArrow: (
      <Box component="div">
        <ArrowForwardIosIcon />
      </Box>
    ),
    responsive: [
      {
        breakpoint: 1280,
        settings: {
          slidesToShow: 3,
        },
      },
      {
        breakpoint: 767,
        settings: {
          slidesToShow: 2,
        },
      },
      {
        breakpoint: 480,
        settings: {
          slidesToShow: 1,
        },
      },
    ],
  };

  var banner = {
    dots: true,
    infinite: true,
    speed: 500,
    slidesToShow: 1,
    slidesToScroll: 1,
    initialSlide: 0,
    arrows: true,
    prevArrow: (
      <Box component="div">
        <ArrowBackIosIcon />
      </Box>
    ),
    nextArrow: (
      <Box component="div">
        <ArrowForwardIosIcon />
      </Box>
    ),
    responsive: [
      {
        breakpoint: 1023,
        settings: {
          arrows: false,
        },
      },
    ],
  };

  function ListItemLink(props) {
    return <ListItem button component="a" {...props} />;
  }

  return (
    <Box component="div" className="main-content">
      <Container maxWidth="lg">
        <Grid container spacing={0}>
          <MediaQuery minWidth={1024}>
            <Grid container item xs={12} md={3} spacing={0}>
              <Box component="div" className="category-menu">
                <Typography component="h3">Categories</Typography>
                <List component="nav">
                  <ListItemLink component={RouterLink} to="/products">
                    <ListItemText primary="Electronic Devices" />
                  </ListItemLink>

                  <ListItemLink component={RouterLink} to="/products">
                    <ListItemText primary="Electronic Accessories" />
                  </ListItemLink>

                  <ListItemLink component={RouterLink} to="/products">
                    <ListItemText primary="TV & Home Appliances" />
                  </ListItemLink>

                  <ListItemLink component={RouterLink} to="/products">
                    <ListItemText primary="Health &amp; Beauty" />
                  </ListItemLink>

                  <ListItemLink component={RouterLink} to="/products">
                    <ListItemText primary="Babies &amp; Toys" />
                  </ListItemLink>

                  <ListItemLink component={RouterLink} to="/products">
                    <ListItemText primary="Groceries &amp; Pets" />
                  </ListItemLink>

                  <ListItemLink component={RouterLink} to="/products">
                    <ListItemText primary="Home &amp; Living" />
                  </ListItemLink>

                  <ListItemLink component={RouterLink} to="/products">
                    <ListItemText primary="Women's Fashion" />
                  </ListItemLink>

                  <ListItemLink component={RouterLink} to="/products">
                    <ListItemText primary="Men's Fashion" />
                  </ListItemLink>

                  <ListItemLink component={RouterLink} to="/products">
                    <ListItemText primary="Fashion Accessories" />
                  </ListItemLink>

                  <ListItemLink component={RouterLink} to="/products">
                    <ListItemText primary="Sports &amp; Lifestyle" />
                  </ListItemLink>

                  <ListItemLink component={RouterLink} to="/products">
                    <ListItemText primary="Automotive &amp; Motorcycles" />
                  </ListItemLink>
                </List>
              </Box>
            </Grid>
          </MediaQuery>
          <Grid container item xs={12} md={9} spacing={0}>
            <Box component="div" className="main-banner">
              <Slider {...banner}>{bannerList}</Slider>
            </Box>
          </Grid>
        </Grid>

        <Box component="div" className="products-box">
          <Typography component="h3">Your Daily Pickups</Typography>
          <Slider {...productListSlider}>{productList}</Slider>
        </Box>

        <Box component="div" className="products-box">
          <Typography component="h3">Best Seller</Typography>
          <Slider {...productListSlider}>{productList}</Slider>
        </Box>

        <Button
          variant="contained"
          color="primary"
          disableElevation
          type="button"
          onClick={() => window.location.replace("/products")}
        >
          See All Products
        </Button>

        <Box className="payment-info-box">
          <Grid container spacing={0}>
            <Grid item xs={12} sm={4} spacing={0}>
              <Box className="box no-border">
                <Avatar
                  alt="Secure Payments"
                  src="/static/images/avatar/1.jpg"
                />
                <Typography component="h3">100% Secure Payments</Typography>
                <Typography>
                  Moving your card details to a{" "}
                  <span> much more secured place.</span>
                </Typography>
              </Box>
            </Grid>
            <Grid item xs={12} sm={4} spacing={0}>
              <Box className="box">
                <Avatar alt="Trustpay" src="/static/images/avatar/1.jpg" />
                <Typography component="h3">Trustpay</Typography>
                <Typography>
                  100% Payment Protection. <span>Easy Return Policy</span>
                </Typography>
              </Box>
            </Grid>
            <Grid item xs={12} sm={4} spacing={0}>
              <Box className="box">
                <Avatar
                  alt="Shop on the Go"
                  src="/static/images/avatar/1.jpg"
                />
                <Typography component="h3">Shop on the Go</Typography>
                <Typography>
                  100% Payment Protection. <span>Easy Return Policy</span>
                </Typography>
              </Box>
            </Grid>
          </Grid>
        </Box>
      </Container>
    </Box>
  );
}
