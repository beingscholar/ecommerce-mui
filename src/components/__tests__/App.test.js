import React from 'react'
import {render, cleanup} from '@testing-library/react';
import App from '../../App';
import Landing from '../../components/Landing';


 afterEach(cleanup)

 describe('app renders', ()=>{
   it('should show landing', () => {
      const { asFragment } = render(<Landing />)
      
      expect(asFragment(<App />)).toMatchSnapshot()
     })
 })
 
 /* it('should take a snapshot', () => {
    const { asFragment } = render(<App />)
    
    expect(asFragment(<App />)).toMatchSnapshot()
   }) */