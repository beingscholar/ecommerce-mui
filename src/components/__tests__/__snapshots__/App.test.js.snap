// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders should show landing 1`] = `
<DocumentFragment>
  <div>
    <header>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <div
            style="display: flex; justify-content: space-between; width: 100%;"
          >
            <div>
              <h6
                class="MuiTypography-root MuiTypography-h6"
              >
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-title-2 MuiTypography-colorInherit"
                  href="/"
                >
                  Home
                </a>
              </h6>
            </div>
            <div
              class="makeStyles-authLinks-3"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6"
              >
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-navLink-1 MuiTypography-colorInherit"
                  href="/signup"
                >
                  Sign Up
                </a>
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-navLink-1 MuiTypography-colorInherit"
                  href="/signin"
                >
                  Sign In
                </a>
              </h6>
            </div>
          </div>
        </div>
      </header>
    </header>
    <main>
      <div
        class="makeStyles-heroContent-4"
        style="height: 80vh;"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
        >
          Your Website
        </h1>
        <p
          class="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary MuiTypography-paragraph MuiTypography-alignCenter"
        >
          This is the admin portion of your website. Keep track of Customers, Products, etc.
        </p>
        <div
          class="makeStyles-heroButtons-5"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <a
                aria-disabled="false"
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                href="/signup"
                tabindex="0"
              >
                <span
                  class="MuiButton-label"
                >
                  Create an Account
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer>
      <div
        class="makeStyles-root-221"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1"
        >
          Copyright Â© Your Website 2020.
        </h6>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
